File: app/src/main/java/com/bunbeauty/ideal/myapplication/adapters/SubscriptionElement.java
package com.bunbeauty.ideal.myapplication.adapters;

import android.content.Context;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import androidx.annotation.NonNull;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.android.ideal.myapplication.R;
import com.bunbeauty.ideal.myapplication.cleanArchitecture.models.entity.User;
import com.bunbeauty.ideal.myapplication.helpApi.SubscriptionsApi;
import com.bunbeauty.ideal.myapplication.helpApi.WorkWithLocalStorageApi;
import com.bunbeauty.ideal.myapplication.cleanArchitecture.data.db.DBHelper;
import com.bunbeauty.ideal.myapplication.cleanArchitecture.ui.activities.profile.ProfileActivity;

public class SubscriptionElement implements View.OnClickListener {

    private static final String OWNER_ID = "owner id";

    private TextView nameText;
    private TextView subscribeText;
    private TextView unsubscribeText;
    private ImageView avatarImage;
    private Context context;
    private View view;

    private String workerId;
    private String workerName;

    public SubscriptionElement(User user, View view, Context context) {
        workerId = user.getId();
        workerName = user.getName();
        this.context = context;
        this.view= view;
    }

    void createElement(){
        onViewCreated(view);
    }


    private void onViewCreated(@NonNull View view) {

        nameText = view.findViewById(R.id.workerNameSubscriptionElementText);
        subscribeText = view.findViewById(R.id.subscribeSubscriptionElementText);
        avatarImage = view.findViewById(R.id.avatarSubscriptionElementImage);
        unsubscribeText = view.findViewById(R.id.unsubscribeSubscriptionElementText);

        nameText.setOnClickListener(this);
        avatarImage.setOnClickListener(this);
        subscribeText.setOnClickListener(this);
        unsubscribeText.setOnClickListener(this);

        LinearLayout layout = view.findViewById(R.id.subscriptionElementLayout);

        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
        params.setMargins(10, 10, 10, 10);
        layout.setLayoutParams(params);

        setData();
    }

    private void setData() {
        nameText.setText(workerName);
        subscribeText.setTag(true);

        DBHelper dbHelper = new DBHelper(context);
        SQLiteDatabase database = dbHelper.getReadableDatabase();

        WorkWithLocalStorageApi workWithLocalStorageApi = new WorkWithLocalStorageApi(database);

        int width = context.getResources().getDimensionPixelSize(R.dimen.photo_avatar_width);
        int height = context.getResources().getDimensionPixelSize(R.dimen.photo_avatar_height);
        workWithLocalStorageApi.setPhotoAvatar(workerId, avatarImage, width, height);
    }

    @Override
    public void onClick(View v) {
        SubscriptionsApi subsApi = new SubscriptionsApi(workerId, context);

        switch (v.getId()) {
            case R.id.subscribeSubscriptionElementText:
                unsubscribeText.setVisibility(View.VISIBLE);
                subscribeText.setVisibility(View.GONE);
                subsApi.unsubscribe();
                Toast.makeText(context, "Подписка отменена", Toast.LENGTH_SHORT).show();
                break;

            case R.id.unsubscribeSubscriptionElementText:
                unsubscribeText.setVisibility(View.GONE);
                subscribeText.setVisibility(View.VISIBLE);
                subsApi.subscribe();
                Toast.makeText(context, "Вы подписались", Toast.LENGTH_SHORT).show();
                break;

            default:
                goToProfile();
                break;
        }
    }

    private void goToProfile() {
        Intent intent = new Intent(context, ProfileActivity.class);
        intent.putExtra(OWNER_ID, workerId);

        context.startActivity(intent);
    }
}

File: app/src/main/java/com/bunbeauty/ideal/myapplication/cleanArchitecture/adapters/elements/SubscriptionElement.kt
package com.bunbeauty.ideal.myapplication.cleanArchitecture.adapters.elements

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.view.View
import com.android.ideal.myapplication.R
import com.bunbeauty.ideal.myapplication.cleanArchitecture.business.CircularTransformation
import com.bunbeauty.ideal.myapplication.cleanArchitecture.data.db.models.entity.Subscriber
import com.bunbeauty.ideal.myapplication.cleanArchitecture.data.db.models.entity.Subscription
import com.bunbeauty.ideal.myapplication.cleanArchitecture.data.db.models.entity.User
import com.bunbeauty.ideal.myapplication.cleanArchitecture.mvp.activities.profile.ProfileActivity
import com.bunbeauty.ideal.myapplication.cleanArchitecture.mvp.presenters.SubscriptionsPresenter
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.subscription_element.view.*

class SubscriptionElement(
    private val view: View,
    private val context: Context,
    private val subscriptionsPresenter: SubscriptionsPresenter,
    private val subscription: Subscription
) {

    fun createElement() {
        view.workerNameSubscriptionElementText.text =
            "${subscription.subscriptionUser.name} ${subscription.subscriptionUser.surname}"
        showAvatar(subscription.subscriptionUser, view)

        view.subscriptionElementLayout.setOnClickListener { goToProfile(subscription.subscriptionUser) }

        view.subscribeSubscriptionElementCheckBox.isChecked = true
        view.subscribeSubscriptionElementCheckBox.setOnCheckedChangeListener { _, isChecked ->
            if (!isChecked) {
                unsubscribe()
            }
        }
    }

    private fun showAvatar(user: User, view: View) {
        val width = context.resources.getDimensionPixelSize(R.dimen.photo_avatar_width)
        val height = context.resources.getDimensionPixelSize(R.dimen.photo_avatar_height)

        Picasso.get()
            .load(user.photoLink)
            .resize(width, height)
            .centerCrop()
            .transform(CircularTransformation())
            .into(view.avatarSubscriptionElementImage)
    }

    private fun unsubscribe() {
        subscriptionsPresenter.deleteSubscription(subscription)
        subscriptionsPresenter.deleteSubscriber(Subscriber(
                userId = subscription.subscriptionId,
                subscriberId = subscription.userId
            )
        )
        //subscriptionsPresenter.deleteUser(user)
    }

    private fun goToProfile(user: User) {
        val intent = Intent(context, ProfileActivity::class.java)
        intent.putExtra(User.USER, user)
        context.startActivity(intent)
        (context as Activity).overridePendingTransition(0, 0)
    }
}